src/router/db.ts

```diff
@@ -202,6 +208,112 @@ db.post('/run/:dbAlias', async (req: Request, res: Response) => {
+  for (const name of moduleNames) {
+    const mod = (Object.values(AllModules) as ModuleInterface[]).find(
+      m => `${m.name}@${m.version}` === name,
+    ) as ModuleInterface;
+    if (!mod) throwError(`This should be impossible. Cannot find module ${name}`);
+    const moduleContext = mod?.provides?.context ?? {};
+    Object.keys(moduleContext).forEach(k => {
+      if (typeof moduleContext[k] === 'function') {
+        context[k] = moduleContext[k];
+      } else {
+        context[k] = cloneDeep(moduleContext[k]);
+      }
+    });
+  }
+  return context;
+}
+
+db.post('/rpc', async (req: Request, res: Response) => {
+  logger.info('Calling /rpc');
+  const { dbId, dbUser, params, modulename, methodname } = req.body;
+  const missing = [];
+  if (!dbId) missing.push('dbId');
+  if (!dbUser) missing.push('dbUser');
+  if (!params) missing.push('params');
+  if (!modulename) missing.push('modulename');
+  if (!methodname) missing.push('methodname');
+  if (!!missing.length) return res.status(400).json(`Required key(s) ${missing.join(', ')} not provided`);
+  try {
+    let output: string | undefined;
+    let error;
+    const user = await MetadataRepo.getUserFromDbId(dbId);
+    const uid = user?.id;
+    const email = user?.email;
+    const dbAlias = user?.iasqlDatabases?.[0]?.alias;
+    const dbRec = await MetadataRepo.getDbById(dbId);
+    const versionString = await TypeormWrapper.getVersionString(dbId);
+    const conn = await TypeormWrapper.createConn(dbId, { name: uuidv4() });
+    // Do not call RPCs if db is upgrading.
+    if (dbRec?.upgrading) {
+      throwError(`Database ${dbId} is upgrading.`);
+    }
+    try {
+      // Look for the Module's instance name with the RPC to be called
+      if (versionString !== config.version) throwError(`Unsupported version ${versionString}`);
+      const [moduleInstanceName] = Object.entries(Modules ?? {})
+        .filter(([_, m]: [string, any]) => m instanceof ModuleBase)
+        .find(([_, m]: [string, any]) => m.name === modulename) ?? ['unknown', undefined];
+      if (!(Modules as any)[moduleInstanceName]) throwError(`Module ${modulename} not found`);
+      const context = await getContext(conn, Modules);
+      const rpcRes: any[] | undefined = await (
+        (Modules as any)[moduleInstanceName] as ModuleInterface
+      )?.rpc?.[methodname].call(dbId, dbUser, context, ...params);
+      output = JSON.stringify(rpcRes);
+      return res.status(200).json(rpcRes);
+    } catch (e) {
+      let errorMessage: string | string[] = logErrSentry(e, uid, email, dbAlias);
+      // split message if multiple lines in it
+      if (errorMessage.includes('\n')) errorMessage = errorMessage.split('\n');
+      // error must be valid JSON as a string
+      error = JSON.stringify({ message: errorMessage });
+      return res.status(400).json({ message: errorMessage });
+    } finally {
+      try {
+        const recordCount = await iasql.getDbRecCount(conn);
+        const rpcCount = 0;
```

Comment:
why are we zero-ing this out now and not simply incrementing it?

Reply:
Oh, because I figured a follow-up PR would drop it? Now that we don't have a table holding onto the RPCs there's no impact on DB performance or disk usage that the rpc count could cause, anymore, so I figured it was dead weight.

Reply:
it is not quite dead weight. it is used here https://github.com/iasql/iasql-engine/blob/main/src/services/telemetry.ts#L18. we only need to update it here https://github.com/iasql/iasql-engine/blob/main/src/entity/index.ts#L40 even if we are dropping the `iasql_rpc` table

Reply:
Discussed offline. The RPC tracking needs to be reworked with the new Postgres-oriented approach combined with the cron job making RPC calls every ~2 minutes. We can also make it better (faster, stronger) so this will be addressed in a follow-up PR.